rest/accounts/v1/credentials_public_keys.go

```diff
@@ -152,6 +154,48 @@ func (c *ApiService) ListCredentialPublicKey(params *ListCredentialPublicKeyPara
 	return ps, err
 }
 
+//Retrieve a single page of CredentialPublicKey records from the API. Request is executed immediately.
+func (c *ApiService) CredentialPublicKeyPage(params *ListCredentialPublicKeyParams, pageToken string, pageNumber string) (*client.Page, error) {
+	path := "/v1/Credentials/PublicKeys"
+
+	data := url.Values{}
+	headers := make(map[string]interface{})
+
+	if params != nil && params.PageSize != nil {
+		data.Set("PageSize", fmt.Sprint(*params.PageSize))
+	}
+
+	data.Set("PageToken", pageToken)
+	data.Set("PageNumber", pageNumber)
+
+	response, err := c.requestHandler.Get(c.baseURL+path, data, headers)
+	if err != nil {
+		return nil, err
+	}
+
+	return client.NewPage(c.baseURL, response), nil
+}
+
+//Streams CredentialPublicKey records from the API as a channel stream. This operation lazily loads records as efficiently as possible until the limit is reached.
+func (c *ApiService) CredentialPublicKeyStream(params *ListCredentialPublicKeyParams, limit int) (chan map[string]interface{}, error) {
+	params.SetPageSize(c.requestHandler.ReadLimits(params.PageSize, limit))
+	page, err := c.CredentialPublicKeyPage(params, "", "")
+	if err != nil {
+		return nil, err
+	}
+	return c.requestHandler.Stream(page, limit, 0), nil
+}
+
+//Lists CredentialPublicKey records from the API as a list. Unlike stream, this operation is eager and will loads 'limit' records into memory before returning.
+func (c *ApiService) CredentialPublicKeyList(params *ListCredentialPublicKeyParams, limit int) ([]interface{}, error) {
```

Comment:
I think this should be refactored into `ListCredentialPublicKey` above (not sure about naming of `List/Stream` prefix or suffix) and should return an array of `ListCredentialPublicKeyResponse`.

So you'd potentially have `(Page/List/Stream)CredentialPublicKey` and callers would expect to find instances of `ListCredentialPublicKeyResponse` instead of a `map`.

Reply:
Used the `baseName`+List/Stream/Page as the func name. Now `Page` and `List` return instances of `ListxxxResponse` but it's a little tricky with `Stream`. Decided to return `chan interface{}` that could then be converted to the appropriate type by the caller. Let me know your thoughts.